ymer/src/main/java/com/avanza/ymer/MirroredObject.java

```diff
@@ -137,10 +138,21 @@ private void setRoutingKey(Document document, T spaceObject) {
 		}
 	}
 
-	private void setInstanceId(Document document, @Nullable Integer instanceId) {
-		if (instanceId != null) {
-			document.put(DOCUMENT_INSTANCE_ID, instanceId);
-		}
+	private void setInstanceIdFields(Document document, InstanceMetadata metadata) {
+		// set current instance id for current amount of partitions
+		metadata.getInstanceId().ifPresent(instanceId -> {
+			metadata.getNumberOfInstances().ifPresent(numberOfInstances ->
+					document.put(getInstanceIdFieldName(numberOfInstances), instanceId)
```

Comment:
For readability, an option is to call the `getInstanceId` method here as well (as on line 152) even though the resulting number should be the same value as `instanceId`

Reply:
Nice suggestion! In that case we could remove `instanceId` from the `InstanceMetadata` object and always calculate it... That would remove some code as well.

Reply:
I changed this now, and I think the resulting code is clearer (though maybe a tiny bit less efficient as we now calculate the current instanceId instead of using the one from the space name?). I could also remove some unused code when we remove fetching instanceId from the space name 👍